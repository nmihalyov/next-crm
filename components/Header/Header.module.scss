.header {
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 25px;
  background-color: #fff;
  border-bottom: 1px solid #eee;
}

.nav {
  display: flex;
}

.navItem {
  position: relative;
  margin-left: 15px;
  &:before {
    content: '';
    position: absolute;
    bottom: -2px;
    left: 50%;
    width: 0;
    height: 1px;
    background-color: #000;
    transition: all .3s ease;
  }
  &:hover:before {
    left: 0;
    width: 100%;  
  }
}