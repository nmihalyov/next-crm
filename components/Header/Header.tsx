import React from 'react';
import Image from 'next/image';
import Link from 'next/link';

import styles from './Header.module.scss';

const Header: React.FC = () => {
  const links = [
    {label: 'Main', href:'/'},
    {label: 'Tasks', href:'/tasks'},
    {label: 'Posts', href:'/posts'},
    {label: 'Users', href:'/users'},
    {label: 'About', href:'/about'}
  ];

  return (
    <header className={styles.header}>
      <Image src='/vercel.svg' alt='logo' width={72} height={16} />
      <nav className={styles.nav}>
        {links.map(el => 
          <Link key={el.href} href={el.href}>
            <a className={styles.navItem}>{el.label}</a>
          </Link>
        )}
      </nav>
    </header>
  );
};

export default Header;