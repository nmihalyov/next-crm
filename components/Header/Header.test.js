import Header from './Header';

describe('Header:', () => {
  it('renders Header', () => {
    const component = shallow(<Header />);

    expect(component).toMatchSnapshot();
  });
});