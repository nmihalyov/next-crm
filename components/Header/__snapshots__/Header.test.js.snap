// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header: renders Header 1`] = `
<Header
  className="header"
>
  <Row
    align="middle"
    justify="space-between"
  >
    <Title
      className="title"
      level={4}
      style={
        Object {
          "fontWeight": 700,
        }
      }
    >
      <ForwardRef(ShareAltOutlined) />
       Next CRM
    </Title>
    <Row
      align="middle"
    >
      <Menu
        mode="horizontal"
        selectedKeys={
          Array [
            undefined,
          ]
        }
        theme="dark"
      >
        <MenuItem
          icon={<ForwardRef(HomeOutlined) />}
          key="/"
          onClick={[Function]}
        >
          Main
        </MenuItem>
        <MenuItem
          icon={<ForwardRef(ContainerOutlined) />}
          key="/tasks"
          onClick={[Function]}
        >
          Tasks
        </MenuItem>
        <MenuItem
          icon={<ForwardRef(BookOutlined) />}
          key="/posts"
          onClick={[Function]}
        >
          Posts
        </MenuItem>
        <MenuItem
          icon={<ForwardRef(UserOutlined) />}
          key="/users"
          onClick={[Function]}
        >
          Users
        </MenuItem>
        <MenuItem
          icon={<ForwardRef(InfoCircleOutlined) />}
          key="/about"
          onClick={[Function]}
        >
          About
        </MenuItem>
      </Menu>
      <Button
        ghost={true}
        style={
          Object {
            "marginLeft": 20,
          }
        }
      >
        Logout
      </Button>
    </Row>
  </Row>
</Header>
`;
